File: server/src/main/java/org/hkurh/doky/controllers/DocumentRequest.java
package org.hkurh.doky.controllers;

public class DocumentRequest {

    private String name;
    private String description;

    public DocumentRequest() {
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }
}

File: server/src/main/kotlin/org/hkurh/doky/documents/api/DocumentRequest.kt
package org.hkurh.doky.documents.api

import jakarta.validation.constraints.NotBlank

/**
 * Represents a document request.
 *
 * @property name The name of the document.
 * @property description The description of the document.
 */
class DocumentRequest {
    @NotBlank(message = "Document Name can not be blank and should contain at least one non space character")
    var name: String = ""
    var description: String = ""
}
